### ISSUE TITLE
Missing -Wsign-compare warning with enum values

### ISSUE THREAD

**Issue by alexhenrie** [llvm/llvm-project#64783]
```c
#include <stdio.h>

int main()
{
    signed int a = 1;
    enum : signed int { b = 1 };
    unsigned int c = -1;

    if (a < c) // condition is true, -Wsign-compare warns about it
        puts("a < c");

    if (b < c) // condition is true, but no warning
        puts("b < c");

    return 0;
}
```

The compiler should warn about both if statements.

$ clang --version
clang version 15.0.7
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

**Comment by EugeneZelenko** [comment]
Could you please try 17 Release Candidate or `main` branch? https://godbolt.org should be helpful.

**Comment by alexhenrie** [comment]
It's not really a bug, sorry for the noise. See [the corresponding GCC bug report](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=111056) for the explanation.
